From 254388a786a240ead08c0b6d577401108558073d Mon Sep 17 00:00:00 2001
From: Dirk Neukirchen <dirkneukirchen@web.de>
Date: Fri, 8 May 2015 09:19:31 +0200
Subject: [PATCH 121/158] mgetty: fix compile on arm64 and mips64 (octeon)

fix several compile errors:
error: initializer element is not constant
error: (near initialization for ...

spotted by debian #764924 and extended to include mips64
(__mips64__ is not defined by gcc, unsure if __LP64__ is usable)

compile tested only

Signed-off-by: Dirk Neukirchen <dirkneukirchen@web.de>
---
 utils/mgetty/patches/002-fix_arm64_mips64.patch | 11 +++++++++++
 1 file changed, 11 insertions(+)
 create mode 100644 utils/mgetty/patches/002-fix_arm64_mips64.patch

diff --git a/utils/mgetty/patches/002-fix_arm64_mips64.patch b/utils/mgetty/patches/002-fix_arm64_mips64.patch
new file mode 100644
index 0000000..a056f6d
--- /dev/null
+++ b/utils/mgetty/patches/002-fix_arm64_mips64.patch
@@ -0,0 +1,11 @@
+--- a/mgetty.h
++++ b/mgetty.h
+@@ -198,7 +198,7 @@ typedef unsigned char uch;
+  */
+ #if defined(__alpha__) || defined(__sparc64__) || \
+ 	    defined(__ia64__) || defined(__s390x__) || defined(__x86_64__) || \
+-	    defined(__powerpc64__)
++	    defined(__powerpc64__) || defined(__aarch64__) || defined(__mips64)
+ # define PTR_IS_LONG
+ #endif
+ 
-- 
2.4.0

